### ISSUE TITLE
Re-implement 'Exit Immediately' Option

### ISSUE THREAD

**Issue by brendanhay** [urbit/urbit#1821]
It appears that sometime in the past during the refactor of `vere/loop.c` circa v0.7.*, the `-x` option's to immediately exit after boot was removed.

Previously the main event loop code looked something like (v0.7.2):

```c
void
u3_lo_loop()
{
  ...
  _lo_init();

  u3_raft_init();

  if ( _(u3_Host.ops_u.tex) ) {
    u3t_boff();
    u3t_damp();
    u3_lo_exit();

    fprintf(stderr, "dry run: exit\r\n");
    exit(0);
  }
  else {
    if ( c3n == u3_Host.ops_u.bat ) {
      uv_run(u3L, UV_RUN_DEFAULT);
    }
  }
}
```

Where `.tex` is the `-x` flag.

This resulted in the desirable feature of being able to separate pier creation (and subsequent exit) from the normal boot/resume process. ie.

1. specify a pill, key file, and pier, initialise and then exit on success or error.
2. specify the pier, and boot/resume into the dojo.

I'd like to go about re-implementing `-x`, does anyone have objections, pointers, or suggestions?

**Comment by joemfb** [comment]
@brendanhay, I implemented a version of this in #1805. The "boot completed" predicate could use some refinement (it currently uses the same condition as the daemon mode fork/exit: configuration of the http server(s)), but it should be sufficient for your purposes. Please let me know if there are any issues.

**Comment by brendanhay** [comment]
@joemfb :+1: I'll give it a spin and work forward from there, thanks!
